compact.go

```diff
@@ -18,18 +18,24 @@ var (
 	errNotSupported          = errors.New("not supported: n>63 encountered when decoding a compact-encoded uint")
 )
 
-type Compact U128
+type BigNumbers interface {
```

Comment:
I see that all numbers implement it, wouldn't it be better to use a more generic name, like "Numeric" or something else?